File: PerfTester-Xamarin/PerfTester-Xamarin/Models/UserDefaultsLoadTest.cs
using System;
using Foundation;

namespace PerfTesterXamarin.Models
{
    public class UserDefaultsLoadTest : Test
    {
        public override string Title => "User Defaults Loading Test";
        public override string Desc => "Loading integers from user defaults";
        public override string ImageName => "Lock";
        public override double[] Parameters => new double[] { 100, 200, 500, 1000, 2000, 5000 };

        public override void Prepare(double param)
        {
            NSUserDefaults uD = NSUserDefaults.StandardUserDefaults;
            uD.RemovePersistentDomain(NSBundle.MainBundle.BundleIdentifier);
            for (long i = 0; i < param; i++)
            {
                uD.SetInt((int)i, String.Format("UD_TEST_{0}", i));
            }
            uD.Synchronize();
        }

        public override void Done(double param)
        {
            NSUserDefaults.StandardUserDefaults.RemovePersistentDomain(NSBundle.MainBundle.BundleIdentifier);
        }

        public override void DoJob(double param)
        {
            NSUserDefaults uD = NSUserDefaults.StandardUserDefaults;
            for (long i = 0; i < param; i++)
            {
                uD.IntForKey(String.Format("UD_TEST_{0}", i));
            }
            base.FinishJob(param);
        }
    }
}

File: PerfTester-Xcode/PerfTester-Xcode/Models/UserDefaultsLoadTest.swift
//
//  UserDefaultsLoadTest.swift
//  PerfTester-Xcode
//
//  Created by Patryk Średziński on 06/11/2018.
//  Copyright © 2018 private.perftesterxcode. All rights reserved.
//

import UIKit

class UserDefaultsLoadTest: Test {
    
    override init() {
        super.init()
        title = "User Defaults Loading Test"
        desc = "Loading integers from user defaults"
        imageName = "Lock"
        parameters = [ 100, 200, 500, 1000, 2000, 5000 ]
    }
    
    override func prepare(param: Double) {
        let uD = UserDefaults.standard
        uD.removePersistentDomain(forName: Bundle.main.bundleIdentifier ?? "")
        for i in 0...Int(param) {
            uD.set(i, forKey: "UD_TEST_\(i)")
        }
        uD.synchronize()
    }
    
    override func done(param: Double) {
        UserDefaults.standard.removePersistentDomain(forName: Bundle.main.bundleIdentifier ?? "")
    }
    
    override func doJob(param: Double) {
        let uD = UserDefaults.standard
        for i in 0...Int(param) {
            uD.integer(forKey: "UD_TEST_\(i)")
        }
        super.finishJob(param: param)
    }
}
